packages/oauth/src/providers/linkedin.ts

```diff
@@ -104,24 +96,26 @@ export const linkedin = <_Auth extends Auth>(auth: _Auth, config: Config) => {
 	} as const satisfies OAuthProvider;
 };
 
+type Locale = {
+	country: string;
+	language: string;
+}
 type LinkedinProfileResponse = {
-	id: string;
-	localizedFirstName: string;
-	localizedLastName: string;
-	profilePicture: {
-		"displayImage~"?: {
-			elements?: Array<{
-				identifiers?: Array<{
-					identifier?: string;
-				}>;
-			}>;
-		};
-	};
+	sub: string;
+	name: string;
+	email: string;
+	email_verified: boolean;
+	family_name: string;
+	family_name: string;
+	locale: Locale,
+	picture: string | null;
 };
 
 export type LinkedinUser = {
 	id: string;
 	firstName: string;
 	lastName: string;
+	email: string;
```

Comment:
I think `email` can be undefined if the `email` scope isn't provided? (See `getLinkedinUser()`)

```suggestion
	email: string | null;
```

Reply:
![image](https://github.com/pilcrowOnPaper/lucia/assets/3163410/1223a61f-c3ba-4b37-a120-bad078c85cb8)
It's hard to identify which properties are optional since MS don't clearly mention it in this [document](https://learn.microsoft.com/en-us/linkedin/consumer/integrations/self-serve/sign-in-with-linkedin-v2#api-request-to-retreive-member-details).
Do you have any reference source for this?

Reply:
This is purely my guess but the existing of `email` scope makes me think the `email` field can be undefined or an empty string.

Reply:
Sound reasonable 👍

Reply:
Forgot to mention but I think it's better to trust to docs and assume it's an empty string if the `email` scope isn't provided. You can leave this part as is

Reply:
I think leave it as optional is acceptable.